unistd/sys.c

```diff
@@ -60,6 +61,23 @@ static int shebang(const char *path)
 	return fd;
 }
 
+static int checkelf(const char *path)
+{
+	int fd;
+	unsigned char head[4];
+	unsigned char elfhead[4] = { 0x7F, 'E', 'L', 'F' };
+
+	if ((fd = open(path, O_RDONLY)) < 0)
+		return -1;
+
+	if ((read(fd, head, 4) < 0) || memcmp(head, elfhead, 4)) {
```

Comment:
```suggestion
	if ((read(fd, head, 4) < 4) || memcmp(head, elfhead, 4)) {
```
Why give up checking the actual number of bytes read?
Why compare against the ELF header if less than 4 bytes were read.

Reply:
Yes. Good notice. Unfortunately after recent commit to kernel this is handled in kernel so while `checkelf()` addition is being dropped. However thank you for an advice!